### PR TITLE
Booth service feature

### PR DESCRIPTION
**Pull request by JanvanZetten**

### CODE REVIEW

File: Core/Application/Implementation/BoothService.cs [hunk 1]
```diff
@@ -1,50 +1,72 @@
...
 using System.Text;
+using Core.Domain;
 using Core.Entity;
 
 namespace Core.Application.Implementation
 {
     public class BoothService : IBoothService
     {
+
+        private readonly IRepository<Booth> _BoothRepo;
```
Comment (asbamse) [inline Core/Application/Implementation/BoothService.cs]: Should be _boothRepo

File: Core/Application/Implementation/BoothService.cs [hunk 2]
```diff
@@ -1,50 +1,72 @@
...
         {
-            throw new NotImplementedException();
+            return _BoothRepo.GetAll().Where(x => x.Booker == null).Count();
         }
 
         public Booth Create(Booth newBooth)
         {
-            throw new NotImplementedException();
+            newBooth.Id = 0;
+           return _BoothRepo.Create(newBooth);
```
Comment (asbamse) [inline Core/Application/Implementation/BoothService.cs]: Shouldn't the service check for faulty input?

File: Core/Application/Implementation/BoothService.cs [hunk 3]
```diff
@@ -1,50 +1,72 @@
...
+                throw new ArgumentOutOfRangeException("Booth with selected ID was not found.");
+            }
+            return booth;
+
         }
 
         public Booth GetUsersBooking(int userId)
         {
-            throw new NotImplementedException();
+            return _BoothRepo.GetAll().Where(b => b.Booker.Id == userId).FirstOrDefault();
```
Comment (asbamse) [inline Core/Application/Implementation/BoothService.cs]: Could be FirstOrDefault instead of Where

File: XUnitTesting/Booth/CreateBoothTest.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+using Moq;
+using Xunit;
+
+namespace XUnitTesting.Booth
+{
+    public class CreateBoothTest
+    {
+        private Mock<IRepository<Core.Entity.Booth>> mockBoothRepository = new Mock<IRepository<Core.Entity.Booth>>();
+
+        public CreateBoothTest()
```
Comment (asbamse) [inline XUnitTesting/Booth/CreateBoothTest.cs:14]: Empty constructor.

**Review by asbamse — COMMENTED** [review]
Random spacing
No documentation

File: Core/Application/Implementation/BoothService.cs [hunk 5]
```diff
@@ -28,12 +28,14 @@ public int CountAvalibleBooths()
...
-            throw new NotImplementedException();
+            newBooth.Id = 0;
+           return _BoothRepo.Create(newBooth);
         }
 
         public Booth Delete(int id)
         {
-            throw new NotImplementedException();
+            GetById(id);
+            return _BoothRepo.Delete(id);
```
Comment (Applemelon) [inline Core/Application/Implementation/BoothService.cs]: Doesn't throw an exception if booth doesn't exist

**Review by Applemelon — CHANGES_REQUESTED** [review]

**Review by Applemelon — DISMISSED** [review]

**Review by Applemelon — COMMENTED** [review]
Looks fine to me

**Review by Applemelon — APPROVED** [review]

**Pull request merged**